### PR TITLE
Observability empty placeholders (Data Quality + Alerts) using EmptyPlaceholder

### LINKED ISSUE #30077: [Observability 2.0] Adopt EmptyPlaceholder for Data Quality & Alerts empty states
## Summary

The Observability 2.0 screens still render legacy empty-state placeholders (`ErrorPlaceHolder` / `FilterTablePlaceHolder`). Adopt the shared `EmptyPlaceholder` core component (`@openmetadata/ui-core-components`) so the empty states match the 2.0 Release mocks and stay consistent across OSS, Collate, and AI mode.

## Scope (OSS-shared components)

- **Data Quality → Test Cases** (`DataQualityTab`): `blank` variant, with a `hasActiveFilters` prop so copy switches between "No test cases yet" and "No matching test cases".
- **Data Quality → Test Suites** (`TestSuitesTable`): `blank` variant, branching on `hasActiveFilters` + subTab (Table / Bundle / filtered).
- **Alerts** (`ObservabilityAlertsTable`): `features` variant onboarding (Pick a trigger / Choose the destination / Stay ahead) with a `+ New Alert` CTA gated on create permission.
- New `label.*` / `message.*` locale keys added to `en-us.json` and synced across locales.

These are OSS-shared components, so the change also flows to Collate and AI mode (tracked separately in open-metadata/openmetadata-collate).

## Acceptance criteria

- Each screen renders the shared `EmptyPlaceholder` (icon + title + description; onboarding + CTA for Alerts) per the 2.0 mocks.
- Filtered vs no-data copy differs on screens that support filtering.
- Unit tests cover both the no-data and filtered branches.

Addressed by #30050.

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
Fixes #30077

### Describe your changes:

Replaces the legacy empty-state placeholders on the Observability 2.0 Data Quality and Alerts screens with the new `EmptyPlaceholder` core component (`@openmetadata/ui-core-components`), matching the 2.0 Release mocks.

**Screens updated (OSS-shared components, so this also covers Collate + AI mode):**

- **Data Quality → Test Cases** (`DataQualityTab`): `blank` variant. New optional `hasActiveFilters` prop so the copy switches between "No test cases yet" (empty) and "No matching test cases" (filtered). Wired from both callers (`TestCases`, `QualityTab`).
- **Data Quality → Test Suites** (`TestSuitesTable`): `blank` variant, branching on `hasActiveFilters` + `subTab` (Table Suites / Bundle Suites / filtered).
- **Alerts** (`ObservabilityAlertsTable`): `features` variant (3-column onboarding — Pick a trigger / Choose the destination / Stay ahead) with a `+ New Alert` CTA gated on create permission.

New `label.*` / `message.*` locale keys added to `en-us.json` and synced across all locales via `yarn i18n`.

### Type of change:

- [x] Improvement

### Tests:

Unit tests updated/added for each empty state (both no-data and filtered branches) asserting the `empty-placeholder` render and copy:
- `DataQualityTab.test.tsx`, `TestSuitesTable.test.tsx`, `ObservabilityAlertsTable.test.tsx` (new), `ObservabilityAlertsPage.test.tsx`.

### UI screenshots:

Empty states now render the shared `EmptyPlaceholder` (icon + title + description, plus onboarding features/CTA for Alerts) per the 2.0 mocks. Screen recording to be attached.

<details open><summary><h3>Greptile Summary</h3></summary>

This PR updates Observability empty states to use the shared `EmptyPlaceholder` component. The main changes are:

- Replaced legacy placeholders on Data Quality test cases and test suites.
- Added filtered versus unfiltered empty-state copy.
- Updated Alerts onboarding with feature cards, create actions, and permission-aware error handling.
- Added locale strings and updated related unit and Playwright tests.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

This looks safe to merge.

- No blocking issues found in the changed code.

</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx | Updates the Alerts empty state with onboarding features, explicit create-permission gating, and a retryable permission-error placeholder. |
| openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/hooks/useObservabilityAlerts.ts | Tracks resource-permission failures and exposes a retry path for the Alerts empty-state error view. |
| openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/ObservabilityAlertsPage.tsx | Passes resource-permission state and retry handling from the Alerts hook into the table. |

</details>

<sub>Reviews (23): Last reviewed commit: ["fix(i18n): restore main&#39;s upload-documen..."](https://github.com/open-metadata/openmetadata/commit/4a1a0f3811f66e2ac1467c315207a0f367c30367) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=44160791)</sub>

<details><summary><h4>Context used (3)</h4></summary>

- Context used - CLAUDE.md ([source](https://app.greptile.com/getcollate/github/open-metadata/openmetadata/-/custom-context?memory=52f0d9d9-cad1-4e7d-b070-de181a0aa5e7))
- Context used - openmetadata-ui-core-components/CLAUDE.md ([source](https://app.greptile.com/getcollate/github/open-metadata/openmetadata/-/custom-context?memory=41754627-b2bf-474b-8082-f37ef1a07013))
- Context used - AGENTS.md ([source](https://app.greptile.com/getcollate/github/open-metadata/openmetadata/-/custom-context?memory=ef262f5f-911f-44f3-8d2d-e9cb1579c8c8))
</details>

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx [hunk 1]
```diff
@@ -107,57 +140,79 @@ function ObservabilityAlertsTable({
     );
   };
 
+  // Hold the empty state until the alerts request AND the resource-permission
+  // fetch (tracked by loadingCount) have both settled. Otherwise an empty
+  // alerts response can render the onboarding before alertResourcePermission
+  // resolves, hiding the "New Alert" CTA from an authorized user.
+  const isAlertsEmpty = !loading && loadingCount === 0 && alerts.length === 0;
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx:159]: Addressed in c62b808ab4.

The empty onboarding is now gated on `!loading && loadingCount === 0 && alerts.length === 0`. `loadingCount` is incremented for the duration of the `getResourcePermission` fetch, so the empty state is held until the resource permission settles instead of rendering with `hasCreatePermission === false` while `alertResourcePermission` is still `undefined`. That closes the race where an authorized user could land on the empty Alerts state without the **New Alert** CTA. Added a covering test (`should hold the empty placeholder while the resource permission is still loading`).

Resolving this and the earlier duplicate threads for the same finding.

[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx:159]: Addressed in b177987d8f.

`fetchAlertResourcePermission` no longer swallows the error — on a failed `getResourcePermission(EVENT_SUBSCRIPTION)` it now calls `showErrorToast`, so the failure is surfaced (with a refresh to retry) rather than silently degrading to a create-less empty state.

We intentionally keep the CTA hidden while the permission is unknown/failed rather than showing it optimistically: rendering **New Alert** for an unresolved permission would expose an action the user may not be authorized for (the backend would 403). So unknown/failed is treated as "not yet known" + a visible error, distinct from a confirmed create — which matches the guidance here.

File: openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/hooks/useObservabilityAlerts.ts [hunk 2]
```diff
@@ -79,8 +79,10 @@ export function useObservabilityAlerts({
...
       setAlertResourcePermission(permission);
-    } catch {
-      // Error
+    } catch (error) {
+      // Surface the failure instead of silently degrading to a create-less
+      // empty state; the user can refresh to retry the permission fetch.
+      showErrorToast(error as AxiosError);
     } finally {
       setLoadingCount((count) => count - 1);
     }
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/hooks/useObservabilityAlerts.ts:93]: Fully addressed in e05008bebc.

The empty-state now distinguishes an **unknown/failed** permission from an **explicit denial** instead of collapsing both to `false`:

```ts
const isCreateExplicitlyDenied = Boolean(
  alertResourcePermission &&
    !alertResourcePermission.Create &&
    !alertResourcePermission.All
);
const hasCreatePermission = !isCreateExplicitlyDenied;
```

So when `getResourcePermission` rejects and leaves `alertResourcePermission` undefined, the **New Alert** CTA is shown (with the error toast from the previous commit signalling the failure), rather than locking an authorized user out. The CTA is hidden only on a confirmed `{Create:false, All:false}` denial; the backend still enforces authorization on submit. Added a unit test covering the unknown-permission path. Resolving.

File: openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx [hunk 3]
```diff
@@ -50,6 +55,41 @@ function ObservabilityAlertsTable({
...
+  // Hide the create CTA only on an explicit denial. When the resource
+  // permission is unknown — e.g. getResourcePermission failed and left it
+  // undefined — don't lock an authorized user out of creating alerts; show the
+  // CTA and let the backend enforce on submit.
+  const isCreateExplicitlyDenied = Boolean(
+    alertResourcePermission &&
+      !alertResourcePermission.Create &&
+      !alertResourcePermission.All
+  );
+  const hasCreatePermission = !isCreateExplicitlyDenied;
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx]: Good catch — reverted to fail-closed in a89bf034e9.

Showing **New Alert** on an unknown permission did fail open, so `hasCreatePermission` is back to `Boolean(alertResourcePermission?.Create || alertResourcePermission?.All)`: the CTA renders only on an **explicit allow**, never on unknown or denied. We don't advertise a privileged action to a user whose create permission isn't confirmed.\n\nThe earlier lockout concern is covered without failing open: `useObservabilityAlerts` now surfaces an error toast when `getResourcePermission` rejects, so an authorized user sees the failure and can refresh to retry rather than silently losing the CTA. Added a test asserting the CTA stays hidden on an unknown permission. Resolving.

File: openmetadata-ui/src/main/resources/ui/src/pages/DataQuality/DataQualityPage.tsx [hunk 4]
```diff
@@ -30,31 +30,72 @@ import observabilityRouterClassBase from '../../utils/ObservabilityRouterClassBa
...
+      onAddBundleSuite,
+      canCreateTestCase,
+      canCreateBundleSuite,
+    });
+  }, [
+    onAddTestCase,
+    onAddBundleSuite,
+    canCreateTestCase,
+    canCreateBundleSuite,
+  ]);
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataQuality/DataQualityPage.tsx]: Fixed in a0327ab598: `setCreateActions` was added to the effect deps, and the whole `RegisterDataQualityCreateActions` registrar was subsequently removed in favour of a `createActions` prop on `DataQualityProvider`, so the effect no longer exists. Resolved.

File: openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/components/ObservabilityAlertsTable.tsx:159]: Closing note for this permission-CTA thread (and its earlier duplicates): resolved across a few commits — (1) the empty state now waits for the permission fetch to settle via `loadingCount === 0`; (2) a failed `getResourcePermission` is surfaced with an error toast instead of being swallowed; (3) the CTA is fail-closed — shown only on an explicit create allow, never on unknown/denied. Final state in a89bf034e9. Resolved.

File: openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/hooks/useObservabilityAlerts.ts [hunk 5]
```diff
@@ -79,8 +79,10 @@ export function useObservabilityAlerts({
       );
 
       setAlertResourcePermission(permission);
-    } catch {
-      // Error
+    } catch (error) {
+      // Surface the failure instead of silently degrading to a create-less
+      // empty state; the user can refresh to retry the permission fetch.
+      showErrorToast(error as AxiosError);
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/ObservabilityAlertsPage/hooks/useObservabilityAlerts.ts:90]: Fully addressed in e99b98998c — the failure is no longer silent and now has an in-page retry.

You're right that `showErrorToast` suppresses 401 and GET-403, so the toast alone was a no-op for exactly these statuses. So the hook now tracks the failure explicitly:\n\n```ts\n} catch (error) {\n  setHasResourcePermissionError(true);   // not reliant on the (suppressed) toast\n  showErrorToast(error as AxiosError);\n}\n```\n\n`useObservabilityAlerts` exposes `hasResourcePermissionError` + `refetchResourcePermission`, and when the empty state would render **and** the permission fetch failed, the table now shows an explicit error placeholder — **"Something went wrong"** + a **Try Again** button that re-runs `getResourcePermission` — instead of the create-less onboarding. So an authorized user gets a visible error and an in-page retry, no refresh required. Still fail-closed on a real `{Create:false, All:false}` denial. Covered by a unit test (error state renders + Try Again invokes the refetch). Resolving.

**Review by karanh37 — DISMISSED** [review]

**Review by karanh37 — APPROVED** [review]
